### PR TITLE
Fix: Spelling mistake of $date['id']; refrences to unlock instead of mod; varchar length

### PR DESCRIPTION
**Pull request by darth3pio**
- $date['id'] should be $data['id']
- Fixed references to unlocks in several comments and strings.
- varchar length in the database shows that name and longname in game_mod is 24 and 48 respectively, updated validator to reflect this.

### CODE REVIEW

**Comment by darth3pio** [comment]
Curiously, your already commited docker-compose changes showed up in my PR... GitHub, so strange.

**Comment by leojonathanoh** [comment]
> Curiously, your already commited docker-compose changes showed up in my PR... GitHub, so strange.

i think the `docker-compose.yml` changes are here because of the change of co-author. You may need to `git rebase` on source branch against `startersclan/asp` `master` branch, instead of merging the  `startersclan/asp` `master` branch into the source branch.

File: src/ASP/frontend/modules/gamedata/js/game_mods.js [hunk 1]
```diff
@@ -40,12 +40,12 @@
...
                 shortName: {
                     required: true,
-                    maxlength: 32,
+                    maxlength: 24,
                     nameRegex: true
                 },
                 longName: {
                     required: true,
-                    maxlength: 64
+                    maxlength: 48
```
Comment (leojonathanoh) [inline src/ASP/frontend/modules/gamedata/js/game_mods.js:48]: so this exactly matches the db varchar length?

Comment (darth3pio) [reply src/ASP/frontend/modules/gamedata/js/game_mods.js:48]: Yep, schema.sql is set at 24 and 48 for these columns so this is the default.

Comment (leojonathanoh) [reply src/ASP/frontend/modules/gamedata/js/game_mods.js:48]: i think its a good fix for validation of the expected string length.

**Comment by leojonathanoh** [comment]
just needs rebasing on `master` and a force push. Do you need help with how to do this?

**Comment by darth3pio** [comment]
> just needs rebasing on `master` and a force push. Do you need help with how to do this?

This would be welcome.

**Comment by leojonathanoh** [comment]
```
# Add upstream and checkout to its master branch
git remote add upstream https://github.com/startersclan/asp
git fetch upstream 
git checkout upstream/master

# Create a new branch for the PR.
git checkout -b fix-spelling-mistake-xxx

# Cherry-pick the relevant commits
git cherry-pick 5dcb924c45c5b4f414574022e243998a1150977d
git cherry-pick 88ca7f3c84907e089cf6bf89a54cc9dc69059cc2
git cherry-pick b2dc1d4018bc346aa40745fce08f1eda2cf73a20

# Open a PR with fix-spelling-mistake-xxx as the source branch
```

**Comment by leojonathanoh** [comment]
The common github workflow recommends to use temporary branches (e.g. `fix-some-bug-xxx`) from the upstream `master` branch for PRs, instead of using a statically-named branch (in this case, the branch name you are using is `startersclan`). Temporary branches allows opening multiple PRs at the same time, each including only the changes it is making, whereas a static branch commits keep growing only one PR can be opened at any time.

**Comment by darth3pio** [comment]
Learning all the things.

**Pull request closed without merging**